consensus/safety-rules/src/safety_rules_manager.rs

```diff
@@ -92,7 +94,9 @@ impl SafetyRulesManager {
             SafetyRulesService::Serializer => {
                 Self::new_serializer(storage, verify_vote_proposal_signature)
             }
-            SafetyRulesService::Thread => Self::new_thread(storage, verify_vote_proposal_signature),
+            SafetyRulesService::Thread => {
+                Self::new_thread(storage, verify_vote_proposal_signature, config.timeout)
```

Comment:
I'd argue if thread ever timed out, we'd have some serious issues :P, but I can see one argument for moving the value out of RemoteService or making thread have a RemoteService that is just ignores except for the timeout.

Reply:
I don't disagree.  Unfortunately, `Thread` invokes `RemoteService::execute` which is a static function and not a struct function.  And `RemoteService::execute` requires a timeout because it is also used by the potentially network-able SafetyRules.

This... behaviour... is a more or less direct result of the c-style static functions that are very dominant in all of the non-network crates as far as I can tell.